frontend/public/components/dashboards-page/overview-dashboard/health-card.tsx

```diff
@@ -88,42 +93,64 @@ const HealthCard_ = connect(mapStateToProps)(({
     watchURL('healthz', fetchK8sHealth);
     watchAlerts();
 
-    subsystems.filter(plugins.isDashboardsOverviewHealthURLSubsystem).forEach(subsystem => {
-      const { url, fetch } = subsystem.properties;
-      watchURL(url, fetch);
-    });
-    subsystems.filter(plugins.isDashboardsOverviewHealthPrometheusSubsystem).forEach(subsystem => {
-      const { query } = subsystem.properties;
-      watchPrometheus(query);
+    subsystems.forEach((subsystem, index) => {
+      if (isDashboardsOverviewHealthURLSubsystem(subsystem)) {
+        const { url, fetch } = subsystem.properties;
+        watchURL(url, fetch);
+      } else if (isDashboardsOverviewHealthPrometheusSubsystem(subsystem)) {
+        const { query, resource } = subsystem.properties;
+        watchPrometheus(query);
+        if (resource) {
+          watchK8sResource(uniqueResource(resource, index));
+        }
+      }
     });
 
     return () => {
       stopWatchURL('healthz');
       stopWatchAlerts();
 
-      subsystems.filter(plugins.isDashboardsOverviewHealthURLSubsystem).forEach(subsystem =>
-        stopWatchURL(subsystem.properties.url)
-      );
-      subsystems.filter(plugins.isDashboardsOverviewHealthPrometheusSubsystem).forEach(subsystem =>
-        stopWatchPrometheusQuery(subsystem.properties.query)
-      );
+      subsystems.forEach((subsystem, index) => {
+        if (isDashboardsOverviewHealthURLSubsystem(subsystem)) {
+          stopWatchURL(subsystem.properties.url);
+        } else if (isDashboardsOverviewHealthPrometheusSubsystem(subsystem)) {
+          const { query, resource } = subsystem.properties;
+          stopWatchPrometheusQuery(query);
+          if (resource) {
+            stopWatchK8sResource(uniqueResource(resource, index));
+          }
+        }
+      });
     };
-    // TODO: to be removed: use JSON.stringify(flags) to avoid deep comparison of flags object
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [watchURL, stopWatchURL, watchPrometheus, stopWatchPrometheusQuery, watchAlerts, stopWatchAlerts, JSON.stringify(flags)]);
+  },
+  // TODO: to be removed: use JSON.stringify(flags) to avoid deep comparison of flags object
+  /* eslint-disable react-hooks/exhaustive-deps */
+  [
+    watchURL,
+    stopWatchURL,
+    watchPrometheus,
+    stopWatchPrometheusQuery,
+    watchAlerts,
+    stopWatchAlerts,
+    watchK8sResource,
+    stopWatchK8sResource,
+    stopWatchAlerts, JSON.stringify(flags),
```

Comment:
Doesn't ESLint complain about two array elements on the same line?

Reply:
it does not, core does not have such strong checking as plugins.
Moved to new line